### DISCUSSION TITLE
Problem -- FileNotFoundError

### DISCUSSION THREAD

**Discussion by Mirdgard** [Extraltodeus/depthmap2mask#33] [Q&A]
Everything worked earlier, but after I've reinstalled "Stable Diffusion" (Automatic1111), there is an error occurs, and I'm a noob in "Python", so I don't know, what to do (and I can't find anything about it in Google). My "Python" version is 3.10.6 
    I understand, that this line gives an error:  
 _File "C:\prog\Git\stable-diffusion-webui\repositories\midas\midas\backbones\next_vit.py", line 8, in <module>
    file = open("./externals/Next_ViT/classification/nextvit.py", "r")
FileNotFoundError: [Errno 2] No such file or directory: './externals/Next_ViT/classification/nextvit.py'_
 
  What should I instal to make it right? It worked before. Please, tell me, should I reinstall "Python" or "Git", or install something on "Git"?
"Depthmap2mask" was installed using this way: "Copy the url of that repository into the extension tab"

-------------   
 Error completing request
Traceback (most recent call last):
  File "C:\prog\Git\stable-diffusion-webui\modules\call_queue.py", line 45, in f
    res = list(func(*args, **kwargs))
  File "C:\prog\Git\stable-diffusion-webui\modules\call_queue.py", line 28, in f
    res = func(*args, **kwargs)
  File "C:\prog\Git\stable-diffusion-webui\modules\img2img.py", line 150, in img2img
    processed = modules.scripts.scripts_img2img.run(p, *args)
  File "C:\prog\Git\stable-diffusion-webui\modules\scripts.py", line 328, in run
    processed = script.run(p, *script_args)
  File "C:\prog\Git\stable-diffusion-webui\extensions\depthmap2mask\scripts\depth2image_depthmask.py", line 76, in run
    sdmg = module_from_file("depthmap_for_depth2img",'extensions/depthmap2mask/scripts/depthmap_for_depth2img.py')
  File "C:\prog\Git\stable-diffusion-webui\extensions\depthmap2mask\scripts\depth2image_depthmask.py", line 22, in module_from_file
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "C:\prog\Git\stable-diffusion-webui\extensions/depthmap2mask/scripts/depthmap_for_depth2img.py", line 11, in <module>
    from repositories.midas.midas.dpt_depth import DPTDepthModel
  File "C:\prog\Git\stable-diffusion-webui\repositories\midas\midas\dpt_depth.py", line 5, in <module>
    from .blocks import (
  File "C:\prog\Git\stable-diffusion-webui\repositories\midas\midas\blocks.py", line 21, in <module>
    from .backbones.next_vit import (
  File "C:\prog\Git\stable-diffusion-webui\repositories\midas\midas\backbones\next_vit.py", line 8, in <module>
    file = open("./externals/Next_ViT/classification/nextvit.py", "r")
FileNotFoundError: [Errno 2] No such file or directory: './externals/Next_ViT/classification/nextvit.py'**

**Comment by Mirdgard** [comment]
So, I've read this topic https://www.reddit.com/r/StableDiffusion/comments/zu720s/deforum_filenotfounderror_errno_2_no_such_file_or/
  And I think, it's a bug. And because I have the latest version of "Stable Diffusion Automatic1111", I can't fix it. 
   So I've decided just not to use [depthmap2mask](https://github.com/Extraltodeus/depthmap2mask) extension, and do what I need using "Inpainting" mode of "Stable Diffusion" and create mask for this purpose using image-editing software.

**Reply by OEM-IS** [reply]
Ive got the same issue. 

FileNotFoundError: [Errno 2] No such file or directory: './externals/Next_ViT/classification/nextvit.py'

**Reply by thorgal1** [reply]
Same

**Comment by Wasobi23** [comment]
I am also getting the exact same error and installed it in the same way.

**Comment by G-force78** [comment]
MiDas repo has messed it up completely, Ive tried so many fixes and they all result in some sort of error, brings to mind the image of the same signpost pointing in all different directions, I'm done with it, is a real shame as this script was excellent for creating new backgrounds. I hope MiDas get around to sorting it soon.

**Reply by Mirdgard** [reply]
> MiDas repo has messed it up completely, Ive tried so many fixes and they all result in some sort of error, brings to mind the image of the same signpost pointing in all different directions, I'm done with it, is a real shame as this script was excellent for creating new backgrounds. I hope MiDas get around to sorting it soon.

 It's much better to create mask for "Inpainting" using image-editing software. Here you can see how it's done. https://www.youtube.com/watch?v=dnIYTGW1m8w

**Comment by windigo00** [comment]
i think i got it fixed. the path in `repositories\midas\midas\backbones\next_vit.py` on line 12 should be relative to app root.
i changed it to `./repositories/midas/externals/Next_ViT/classification/nextvit.py` and it works.

if there is still something missing, run `.\repositories\midas\install_next_vit.sh`
hope it helps ;)

**Reply by Mirdgard** [reply]
> i think i got it fixed. the path in `repositories\midas\midas\backbones\next_vit.py` on line 12 should be relative to app root. i changed it to `./repositories/midas/externals/Next_ViT/classification/nextvit.py` and it works.
> 
> if there is still something missing, run `.\repositories\midas\install_next_vit.sh` hope it helps ;)

  Very good! Thank you!
